src/main/java/de/tum/in/www1/artemis/repository/StatisticsRepository.java

```diff
@@ -0,0 +1,98 @@
+package de.tum.in.www1.artemis.repository;
+
+import java.time.Instant;
+import java.time.ZonedDateTime;
+import java.util.List;
+import java.util.Map;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+import org.springframework.stereotype.Repository;
+
+import de.tum.in.www1.artemis.domain.User;
+
+/**
+ * Spring Data JPA repository for the user statistics
+ */
+@Repository
+public interface StatisticsRepository extends JpaRepository<User, Long> {
+
+    @Query("""
+            select count(distinct u.login)
+            from User u, PersistentAuditEvent p
+            where u.login like p.principal and p.auditEventType = 'AUTHENTICATION_SUCCESS' and u.login not like '%test%' and p.auditEventDate >= :#{#span}
+            """)
+    Integer getLoggedInUsers(@Param("span") Instant span);
+
+    @Query("""
+            select count(distinct u.login)
+            from User u, Submission s, StudentParticipation p
+            where s.participation.id = p.id and p.student.id = u.id and s.submissionDate >= :#{#span} and u.login not like '%test%'
+            """)
+    Integer getActiveUsers(@Param("span") ZonedDateTime span);
+
+    @Query("""
+            select count(e.id)
+            from Exercise e
+            where e.releaseDate >= :#{#span} and e.releaseDate <= :#{#now}
+            """)
+    Integer getReleasedExercises(@Param("span") ZonedDateTime span, @Param("now") ZonedDateTime now);
+
+    @Query("""
+            select count(e.id)
+            from Exercise e
+            where e.dueDate >= :#{#span} and e.dueDate <= :#{#now}
+            """)
+    Integer getExerciseDeadlines(@Param("span") ZonedDateTime span, @Param("now") ZonedDateTime now);
+
+    @Query("""
+            select count(e.id) from Exam e where e.endDate >= :#{#span} and e.endDate <= :#{#now}
```

Comment:
This one is not formatted like the rest is.

Reply:
[Done](https://github.com/ls1intum/Artemis/pull/2471/commits/098393562c8ad4db3f03a38571867afd437fe9e8)